### PR TITLE
Update functions to return bools rather than ints

### PR DESCRIPTION
**Pull request by Starbuck5**
Someone on [stackoverflow](https://stackoverflow.com/questions/66739461/pygame-colliderect-function-returning-0-zeros-and-1-ones-instead-of-true-a) pointed out that some `Rect` functions return `1`  or `0` instead of the `True` or `False` that the documentation says.

I don't think many people noticed, because of how close `1` and `0` are to `True` and `False`, but these changes up our docs consistency a bit.

**Summary of changes:**

I started this by fixing the inconsistency pointed out about Rect, and then I moved on to other modules. I used grep to find all instances of `return PyInt_FromLong(...)`, and replaced them with `PyBool_FromLong` when appropriate.

One change that might be noticed the most is `key.get_pressed()` returning a list of booleans rather than integers.

~~The most extensive changes were to `display.toggle_fullscreen()`. The docs say that it returns an int, but not what that int means. Looking at the SDL1 toggle_fullscreen function, it should return 0/False on failure, 1/True on success. I edited the SDL2 toggle_fullscreen function to match this, rather than returning `-1` on failure as it seemed to be doing. I also edited the `toggle_fullscreen()` docs to say it returns a boolean, True on success and False on failure.~~

While I had the `toggle_fullscreen()` docs right there, I added a notice about #2380.

Fun fact, I did some digging in the commit history, and some of these lines were written before Python had booleans. 😮

Edit: related to #1673

### CODE REVIEW

**Comment by robertpfeiffer** [comment]
Yeah, that was undocumented on purpose...

File: docs/reST/ref/display.rst [hunk 1]
```diff
@@ -514,6 +515,10 @@ required).
     * wayland (Linux/Unix)
     * cocoa (OSX/Mac)
 
+   .. Note:: :func:`toggle_fullscreen` doesn't work on Windows
+             unless the window size is in :func:`pygame.display.list_modes()`.
+             See `issue #2380 <https://github.com/pygame/pygame/issues/2380>`_.
+
    .. ## pygame.display.toggle_fullscreen ##
```
Comment (robertpfeiffer) [inline docs/reST/ref/display.rst:522]: That only goes for non-SCALED modes, right?

Comment (Starbuck5) [reply docs/reST/ref/display.rst:522]: I hadn't tested with SCALED modes before, but you're right. I'll update that.

**Comment by Starbuck5** [comment]
> Yeah, that was undocumented on purpose... I wasn't sure if toggle_fullscreen should perhaps return the resolution or the display surface in the future, and previously it returned nothing.

The SDL1 one looked like it returned 1 on success and 0 on failure, so I changed them both to return True on success and False on failure. @robertpfeiffer is that alright? If it needs more discussion and I can pull it out of this PR.

**Comment by robertpfeiffer** [comment]
It probably needs a *little* more discussion. Some of these cases should issue a warning or a message of some kind, but you know how it is with warnings and backward compatibility...

On some platforms, toggling fullscreen is just not a thing, and if that is the case, this should be handled by the user. Maybe we need a can_toggle_fullscreen() function (for Android and framebuffer and RPi and KMSDRM), but since that would be retrofitted _after_ togglr_fullscreen was a thing, it would not be backward-compatible either.

**Comment by Starbuck5** [comment]
I see.

The previous output, although undocumented, did give 3 distinct pieces of information; `-1`, unsupported; `0` , failure; or `1`, success.

Adding warnings or adding a `can_toggle_fullscreen()` doesn't block this though, so I'm going to keep my change in for now, unless you would like me to take it out.

I could see a `can_toggle_fullscreen()` function being useful. For example, an options screen that only presents a fullscreen / not fullscreen option if that is an actual option.

**Comment by robertpfeiffer** [comment]
> The previous output, although undocumented, did give 3 distinct pieces of information; `-1`, unsupported; `0` , failure; or `1`, success.

I'll be happy if you add comments like
```C
Py_RETURN_FALSE; //unsupported

Py_RETURN_FALSE; //failed
```

**Comment by Starbuck5** [comment]
I realized that it raised an error in lots of cases, and I didn't want to change those all to return `False`, because that isn't very transparent, so this commit is another stab at making everything consistent.

The `versionchanged` comment assumes pygame 1.9.x people are using SDL1 and pygame 2.x people are using SDL2, which is true for almost everyone.

**Review by illume — CHANGES_REQUESTED** [review]
If I'm understanding correctly, there's still some open discussion on the toggle stuff? If so, maybe we could take the toggle stuff out of this PR so it doesn't block getting merged? Feels like the toggle stuff needs more before it's documented?

**Comment by Starbuck5** [comment]
I took all the toggle stuff out, except for the warning about it not working properly on windows.
I think that's important to keep in.

**Comment by robertpfeiffer** [comment]
I'd be happy if this was commented, or if the toggle stuff was kept out for now, or if we printed a warning. Either is fine.

**Comment by robertpfeiffer** [comment]
So for now, all my objections are withdrawn or have been addressed.

**Comment by Starbuck5** [comment]
> So for now, all my objections are withdrawn or have been addressed.

Is that an approving review @robertpfeiffer? 😉

**Review by illume — APPROVED** [review]
Thanks :)

**Comment by illume** [comment]
Sounds like robertpfeiffer approved, so I'll merge.

**Pull request merged**